contracts/DLT/interfaces/IDLTEnumerable.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "./IDLT.sol";
+
+/**
+ * @title DLT Standard, optional enumeration extension
+ */
+interface IDLTEnumerable is IDLT {
+    /**
+     * @dev Returns the total number of main ids.
+     */
+    function totalMainIds() external view returns (uint256);
+
+    /**
+     * @dev Returns the total number of sub ids for each main Ids.
+     */
+    function totalSubIds(uint256 mainId) external view returns (uint256);
+
+    /**
+     * @dev Returns the total supply of main ids.
+     */
+    function mainTotalSupply(uint256 mainId) external view returns (uint256);
+
+    /**
+     * @dev Returns the total supply of sub ids for each main Ids.
+     */
+    function subTotalSupply(
+        uint256 mainId,
+        uint256 subId
+    ) external view returns (uint256);
+
+    /**
+     * @dev Returns array of all sub ids for a main id
+     */
+    function subIds(uint256 mainId) external view returns (uint256[] memory);
+
+    /**
+     * @dev Returns total sub id balance of owner for each main id
+     */
+    function totalSubIdBalance(
+        address owner,
+        uint256 mainId
+    ) external view returns (uint256);
```

Comment:
may we think for better names ? we can specify names that let the developer specify what does this method do from just the name.